### PR TITLE
Simplify HackerNewsFeedProvider closes #94

### LINKED ISSUE #94: Refactor HackerNewsFeedProvider
**User Story**
As a developer, I would like to refactor `HackerNewsFeedProvider` to simplify the serialisation logic. 

**Acceptance Criteria**

- Unescape HTML entities in the text body.
- Refactor HackerNewsFeedProvider to avoid creating the same HackerNewsData twice
- Moved custom setter logic into HackerNewsData
- Cache WebClient to avoid creating a new one for every request

**Notes**

### PR DESCRIPTION
**Pull request by Minus20Five**
GitHub Issue (If applicable): # #94

## PR Type

What kind of change does this PR introduce?
- Refactoring (no functional changes, no api changes)

## What is the current behavior?

Expected.

## What is the new behavior?

- Unescape HTML entities in the text body.
- Refactored `HackerNewsFeedProvider` to avoid creating the same `HackerNewsData` twice
- Moved custom setter logic into `HackerNewsData`
- Cache `WebClient` to avoid creating a new one for every request.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tested code with current upstream repo
- [ ] ~Added new tests to the test suite for the feature~
- [x] Tested the full test suite, and ensured that the feature does not break current build.
- [ ] ~Docs have been added/updated which fit (for bug fixes / features)~
- [x] Contains **NO** breaking changes
- [ ] ~Associated with an issue (GitHub or internal)~
- [ ] ~Checked licensing of Frameworks/Libraries (if applicable)~

## Other information

Internal Issue (If applicable): N/A

### CODE REVIEW

**Comment by Minus20Five** [comment]
@salma-s 

Hi, I was in the middle of commenting on your PR and then it got merged.
Anyway, these are the changes I was suggesting, please take a look. =)

File: server/src/main/java/com/feeder/server/model/HackerNewsData.java [hunk 1]
```diff
@@ -41,20 +45,34 @@ public Type feedType() {
...
+
+    public abstract Builder setUrl(String url);
+
+    public abstract Builder setScore(Integer score);
 
-    Builder by(String by);
+    public abstract Builder setText(String text);
 
-    Builder time(Integer time);
+    abstract int id();
```
Comment (HarrisonLeach1) [inline server/src/main/java/com/feeder/server/model/HackerNewsData.java]: Should these have visibility modifiers?

Comment (Minus20Five) [reply server/src/main/java/com/feeder/server/model/HackerNewsData.java]: No, they should be package private.

**Comment by Minus20Five** [comment]
> LGTM @Minus20Five, thanks for the refactor. However, all PRs should be in regards to an issue. Can you please write a story for this, and them im happy to approve. (required by assignment)

Done.

**Comment by salma-s** [comment]
A bit delayed but thanks @Minus20Five for the refactor!

**Pull request merged**